### PR TITLE
[AWS Cloudwatch] Changed module to call ListMetrics API only once per region

### LINKED ISSUE #33134: [AWS Cloudwatch] Avoid calling ListMetrics by namespace
Avoid calling ListMetrics by namespace, and do a single call per region instead. Use the API response to filter the namespaces we are interested in.

Full context behind this change is available at https://github.com/elastic/integrations/issues/2913

### PR DESCRIPTION
**Pull request by girodav**
## What does this PR do?

This PR modifies the AWS Cloudwatch module to call ListMetrics API only once per region, instead of once per namespace. Any namespaces that are part of ListMetrics API's response, but not configured to be queried, are removed in code.

## Why is it important?

ListMetrics API is billed per number of calls, so reducing the total number of calls lead to cost savings for users of the AWS Cloudwatch module.

References:
- https://github.com/elastic/beats/issues/33134
- https://github.com/elastic/integrations/issues/2913

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Related issues

- Closes #33134

### CODE REVIEW

**Review by aspacca — APPROVED** [review]

File: x-pack/metricbeat/module/aws/cloudwatch/cloudwatch.go [hunk 1]
```diff
@@ -180,10 +180,22 @@ func (m *MetricSet) Fetch(report mb.ReporterV2) error {
 		if err != nil {
 			m.Logger().Warn("skipping metrics list from region '%s'", regionName)
 		}
+
+		// retrieve all the details for all the metrics available in the current region
+		regionListMetricsOutput, err := aws.GetListMetricsOutput("*", regionName, m.Period, svcCloudwatch)
+
+		// filter out metrics details for namespaces that are not included in the configuration
+		listMetricsOutput := make([]types.Metric, 0)
```
Comment (aspacca) [inline x-pack/metricbeat/module/aws/cloudwatch/cloudwatch.go:188]: it seems we should add an if in `filterListMetricsOutput` in order to be sure that the metrics belong the current namespace we are iterating

**Review by aspacca — COMMENTED** [review]
please see my comment

File: x-pack/metricbeat/module/aws/cloudwatch/cloudwatch.go (hunk 1 above)
Comment (girodav) [reply x-pack/metricbeat/module/aws/cloudwatch/cloudwatch.go:188]: You are right. I made this change in https://github.com/elastic/beats/pull/34055/commits/c58d9d267167a7f8758ac03a63e8b3b6dcb7e8e7 . We could technically remove this part now as the if added in this new commit would filter out also the namespaces we don't have in the conf, but I think it's good to explicitly do it here. What do you think? In terms of performance, it does not change much.

File: x-pack/metricbeat/module/aws/cloudwatch/cloudwatch.go [hunk 2]
```diff
@@ -180,10 +180,13 @@ func (m *MetricSet) Fetch(report mb.ReporterV2) error {
 		if err != nil {
 			m.Logger().Warn("skipping metrics list from region '%s'", regionName)
 		}
+
+		// retrieve all the details for all the metrics available in the current region
+		listMetricsOutput, _ := aws.GetListMetricsOutput("*", regionName, m.Period, svcCloudwatch)
```
Comment (kaiyan-sheng) [inline x-pack/metricbeat/module/aws/cloudwatch/cloudwatch.go:185]:
```suggestion
		listMetricsOutput, err := aws.GetListMetricsOutput("*", regionName, m.Period, svcCloudwatch)
```

Comment (kaiyan-sheng) [reply x-pack/metricbeat/module/aws/cloudwatch/cloudwatch.go:185]: and also check the error?

File: x-pack/metricbeat/module/aws/cloudwatch/cloudwatch.go [hunk 3]
```diff
@@ -180,10 +180,13 @@ func (m *MetricSet) Fetch(report mb.ReporterV2) error {
...
+
 		for namespace, namespaceDetails := range namespaceDetailTotal {
 			m.logger.Debugf("Collected metrics from namespace %s", namespace)
 
-			listMetricsOutput, err := aws.GetListMetricsOutput(namespace, regionName, m.Period, svcCloudwatch)
 			if err != nil {
```
Comment (kaiyan-sheng) [inline x-pack/metricbeat/module/aws/cloudwatch/cloudwatch.go:190]: We can remove the error checking here

File: x-pack/metricbeat/module/aws/cloudwatch/cloudwatch.go [hunk 4]
```diff
@@ -192,9 +195,9 @@ func (m *MetricSet) Fetch(report mb.ReporterV2) error {
 			if len(listMetricsOutput) == 0 {
```
Comment (kaiyan-sheng) [inline x-pack/metricbeat/module/aws/cloudwatch/cloudwatch.go:195]: this can go outside the for loop?

**Review by kaiyan-sheng — APPROVED** [review]

**Pull request merged**